Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,216 @@
+removed or discouraged and in which way the TSFE class itself may be
+interesting to developers.
+
+Accessing TSFE
+===============
+
+.. attention::
+
+    Some of the former public properties and methods have been changed to
+    protected or marked as internal. Often, accessing TSFE is no longer
+    necessary, and there are better alternatives.
+
+    Directly access :php:`$GLOBALS['TSFE']` only as a last resort. It is
+    strongly discouraged if not absolutely necessary.
+
+From the source:
+
+    When calling a frontend page, an instance of this object is available
+    as :php:`$GLOBALS['TSFE']`, even though the Core development strives to get
+    rid of this in the future.
+
+If access to the
+:php:`\TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController` instance is
+necessary, use the request attribute
+:ref:`frontend.controller <typo3-request-attribute-frontend-controller>`:
+
+..  code-block:: php
+
+    $frontendController = $request->getAttribute('frontend.controller');
+
+TSFE is not available in all contexts. In particular, it is
+only available in frontend contexts, not in the backend or CLI.
+
+Initializing :php:`$GLOBALS['TSFE']` in the backend is sometimes done in code
+examples found online. This is not recommended. TSFE is not initialized in the
+backend context by the Core (and there is usually no need to do this).
+
+From the PHP documentation:
+
+    As of PHP 8.1.0, $GLOBALS is now a read-only copy of the global symbol table.
+    That is, global variables cannot be modified via its copy.
+
+https://www.php.net/manual/en/reserved.variables.globals.php
+
+Howtos
+======
+
+.. add howtos for common use cases, remove the not recommended, deprecated
+.. way if there is a better way
+
+Following are some examples which use TSFE and alternatives to using TSFE,
+where available:
+
+.. _tsfe_ContentObjectRenderer:
+
+Access ContentObjectRenderer
+----------------------------
+
+Access the :php:`ContentObjectRenderer` (often referred to as "cObj"):
+
+.. code-block:: php
+
+    // this is discouraged, obtain TSFE from request attribute 'frontend.controller'
+    // see next example
+    $cObj = $GLOBALS['TSFE']->cObj;
+
+.. code-block:: php
+
+    $frontendController = $request->getAttribute('frontend.controller');
+    $cObj = $frontendController->cObj;
+
+In the case of a non Extbase plugin via setter injection:
+
+.. code-block:: php
+
+    public function setContentObjectRenderer(ContentObjectRenderer $cObj): void
+    {
+        $this->cObj = $cObj;
+    }
+
+.. _tsfe_pageId:
+
+Access current page id
+----------------------
+
+Access the current page id:
+
+.. code-block:: php
+
+    // this is discouraged
+    $GLOBALS['TSFE']->id
+
+Can be done using the :ref:`'routing' <typo3-request-attribute-routing>`
+request attribute:
+
+.. code-block:: php
+
+    $pageArguments = $request->getAttribute('routing');
+    $pageId = $pageArguments->getPageId();
+
+.. _tsfe_language:
+
+Access language settings
```

Comment:
I am missing here the discouraged usage of TSFE to obtain the language settings. This may help users migrating from older versions.